File: rev-com-server/src/main/java/com/codergb/bean/Actor.java
package com.codergb.bean;

import lombok.*;

import java.util.Date;

@Data
@NoArgsConstructor
@AllArgsConstructor
@ToString
@EqualsAndHashCode(callSuper = true)
public class Actor extends FileInfo{
  private String id;
  private String name;
  private String avatarUrl;
  private String foreignName;
  private String alias;
  private String constellation;
  private String profession;
  private Date birth;
  private String birthPlace;
  private String family;
  private String description;
  private Date createTime;
  private Date updateTime;
}

File: rev-com-manage/src/types/actor.ts
import {IOccupation} from "@/types/occupation";
import {IFileInfo} from "@/types/FileInfo";

interface IActor extends IFileInfo{
  id:string,
  name:string,
  avatarUrl:string,
  foreignName:string,
  alias:string,
  constellation:string,
  occupations:IOccupation[],
  birth:string,
  birthPlace:string,
  family:string,
  description:string,
  createTime:string,
  updateTime:string,
}
export type {
  IActor
}
